File: 209. Minimum Size Subarray Sum-1.cpp
class Solution {
public:
  int minSubArrayLen(int target, vector<int> &nums) {
    int n = nums.size();

    int i = 0, j = 0, sum = 0, res = n + 1;
    while (j < n) {
      sum += nums[j];
      while (sum >= target) {
        res = min(res, j - i + 1);
        sum -= nums[i++];
      }
      j++;
    }

    return res == n + 1 ? 0 : res;
  }
};

File: 209. Minimum Size Subarray Sum-1.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        res = n + 1
        s = 0
        left = 0
        for right, x in enumerate(nums):
            s += x
            while s >= target:
                res = min(res, right - left + 1)
                s -= nums[left]
                left += 1

        return res if res != n + 1 else 0
